time/src/main/java/net/jqwik/time/internal/properties/arbitraries/valueRanges/ValueRange.java

```diff
@@ -0,0 +1,61 @@
+package net.jqwik.time.internal.properties.arbitraries.valueRanges;
+
+public abstract class ValueRange<T> {
+
+	private T min;
+	private T max;
+
+	protected void exceptionCheck(Parameter parameter) {
+		//Override if needed
+		//do nothing in default case
+	}
+
+	protected void minMaxChanger(Parameter parameter) {
+		//Override if needed
+		//do nothing in default case
+	}
+
+	public ValueRange<T> set(T min, T max) {
+		min = min != null ? min : this.min;
+		max = max != null ? max : this.max;
+		Parameter parameter = new Parameter(min, max);
+		minMaxChanger(parameter);
+		exceptionCheck(parameter);
+		this.min = parameter.min;
+		this.max = parameter.max;
+		return this;
+	}
+
+	public T getMin() {
+		return min;
+	}
+
+	public T getMax() {
+		return max;
+	}
+
+	protected class Parameter {
```

Comment:
As far as I can see, instances of Parameter are only used in methods of subclasses. The class doesn't pull its weight IMO. Indeed it makes the interface of minMaxChanger() and exceptionCheck() less obvious. I'd get rid of it and just use the two parameters min and max.